app/autodimen.gradle.kts

```diff
@@ -0,0 +1,208 @@
+import org.w3c.dom.Element
+import org.w3c.dom.Node
+import java.io.PrintWriter
+import java.util.*
+import javax.xml.parsers.DocumentBuilderFactory
+import kotlin.collections.HashMap
+
+enum class DimenType {
+    /**
+     * Best for auto generate the dimension by the default values/dimens.xml file.
+     */
+    FROM_DIMEN_FILE,
+
+    /**
+     * Best for auto generate the dimension as default.
+     */
+    AUTO_CREATE
+}
+
+open class DimenFactory : DefaultTask() {
+
+    @Input
+    private val dimens = listOf(
+        320,
+        360,
+        384,
+        411,
+        480,
+        540,
+        600,
+        720,
+        800,
+        960,
+        1024,
+        1080,
+        1280,
+        1440,
+        2560,
+        3840
+    )
+
+    @Input
+    private val baseDimen = 360
+
+    @Input
+    private val positiveMaxDP = 500
+
+    @Input
+    private val positiveMaxSP = 60
+
+    @Input
+    private val negativeMaxDP = 180
+
+    @Input
+    private val type = DimenType.AUTO_CREATE
+
+    @Input
+    private val dimenFileName = "values/dimens.xml"
+
+    private val resFolder = project.projectDir.path + "/src/main/res/"
+
+    @TaskAction
+    fun create() {
+        when (type) {
+            DimenType.AUTO_CREATE -> autoCreateDimen()
+            DimenType.FROM_DIMEN_FILE -> createDimenFromDimenFile()
+        }
+    }
+
+    private fun createDimenFromDimenFile() {
```

Comment:
@huytq-2422 a đang k hiểu đã có file dimen theo screen rồi lại cần có các hàm này nhỉ

Reply:
@bacnv-1260 Những hàm này để sinh ra các file dimen theo screen. Pull này em sinh luôn từ đầu. 
Ý của em vẫn giữ lại file này là: 
- Trong quá trình sử dụng, nếu muốn phát sinh thêm dimen nữa thì mình chỉ cần khai báo ở file dimen và rebuild là nó tự sinh ra ở các file khác mà không cần tính toán. 
- Nếu dự án khác lấy về thì sẽ thay đổi type thành DimenType.AUTO_CREATE để nó build từ đầu

Reply:
@huytq-2422
thế nhớ viết readme cho phần này nhé